org.lflang/src/org/lflang/cli/CliBase.java

```diff
@@ -82,136 +113,144 @@ protected CliBase(String toolName) {
         // Main instance.
         final CliBase main = injector.getInstance(toolClass);
-        main.runMain(args);
+        // Parse arguments and execute main logic.
+        CommandLine cmd = new CommandLine(main)
+            .setOut(new PrintWriter(io.getOut()))
+            .setErr(new PrintWriter(io.getErr()));
+        int exitCode = cmd.execute(args);
+        io.callSystemExit(exitCode);
     }
 
+    public abstract void run();
+
     protected static Injector getInjector(String toolName, Io io) {
-        final ReportingBackend reporter = new ReportingBackend(io, toolName + ": ");
+        final ReportingBackend reporter 
+            = new ReportingBackend(io, toolName + ": ");
 
         // Injector used to obtain Main instance.
-        return new LFStandaloneSetup(new LFRuntimeModule(), new LFStandaloneModule(reporter, io))
-            .createInjectorAndDoEMFRegistration();
+        return new LFStandaloneSetup(
+            new LFRuntimeModule(),
+            new LFStandaloneModule(reporter, io)
+        ).createInjectorAndDoEMFRegistration();
     }
 
+    /**
+     * Resolve to an absolute path, in the given {@link #io} context.
+     */
+    protected Path toAbsolutePath(Path other) {
+        return io.getWd().resolve(other).toAbsolutePath();
+    }
 
     /**
-     * Main function of the tool.
+     * Filter the command-line arguments needed by the code generator, and
+     * return them as properties.
      *
-     * @param args Command-line arguments.
+     * @return Properties for the code generator.
      */
-    protected void runMain(final String... args) {
-
-        // Main instance.
-        // Apache Commons Options object configured to according to available CLI arguments.
-        Options options = getOptions();
-        // CLI arguments parser.
-        CommandLineParser parser = new DefaultParser();
-        // Helper object for printing "help" menu.
-        HelpFormatter formatter = new HelpFormatter();
-
-        final Option helpOption = new Option("h", "help", false, BuildParm.HELP.description);
-        final Option versionOption = new Option("version", "version", false, BuildParm.VERSION.description);
-
-        options.addOption(helpOption);
-        options.addOption(versionOption);
-
-        CommandLine cmd;
-        try {
-            cmd = parser.parse(options, args, false);
-        } catch (ParseException e) {
-            reporter.printFatalError(
-                "Unable to parse command-line arguments. Reason: " + e.getMessage() + "\n"
-                    + "The full command-line was: " + Arrays.toString(args));
-            printHelp(options, formatter, io.getErr());
-            io.callSystemExit(1);
-            return;
-        }
-
-        // If requested, print help and abort
-        if (cmd.hasOption(helpOption.getOpt())) {
-            printHelp(options, formatter, io.getOut());
-            io.callSystemExit(0);
-        }
+    protected Properties filterPassOnProps() {
```

Comment:
This appears to be specific to lfc. Why is it located in the base class?

Reply:
True, just fixed it.